AK/Utf16View.h

```diff
@@ -138,7 +138,8 @@ class Utf16View {
     size_t calculate_length_in_code_points() const;
 
     ReadonlySpan<u16> m_code_units;
-    mutable Optional<size_t> m_length_in_code_points;
+    mutable size_t m_length_in_code_points;
+    bool m_has_cached_length_in_code_points { false };
```

Comment:
Just to as a side note:
This is inking the optional to allow reuse of the padding

Same thing can be achieved with `[[no_unique_address]]`

Reply:
Wait what, that's such a nice trick! Let's do that instead :)

Reply:
Well I read this <https://devblogs.microsoft.com/oldnewthing/20240731-00/?p=110069> today, so it's still fresh in my memory